src/odemis/driver/xt_client.py

```diff
@@ -1469,7 +1498,19 @@ def __init__(self, name, role, parent, channel_name, **kwargs):
         self._shape = (256,)  # Depth of the image
         self.data = SEMDataFlow(self)
 
-        self._channel_name = channel_name
+        if hasattr(self.parent, "_scanner") and hasattr(self.parent, "_fib_scanner"):
+            self.active_scanner_type = StringEnumerated("electron", choices={"electron", "ion"},
+                                                        setter=self._set_active_scanner_type)
```

Comment:
Let's call this VA just "scanner".
Also, the content should be the **name** of the components. That's how we link the components in Odemis in general, so let's keep the same type of interface.

Reply:
with name of the component you mean Scanner/FibScanner (as in name of the class) or do you mean the role of the component?

Reply:
I mean _scanner.name or _fib_scanner.name . These are the names picked by the "user" (ie, set in the microscope file).